back/backend/models/province.py

```diff
@@ -0,0 +1,26 @@
+from google.cloud import ndb
+
+from backend.models.region import Region
+
+# Cache mapping of province names.
+provinces_by_name = {}
```

Comment:
Is a cache really necessary? Seems like it's possibly premature optimization and we would benefit more from making the getter simpler.

Reply:
I get your point but I think we have few provinces and a lot of references to them it actually would save us a ton of request to keep the cache .If we remove it we will be querying the datastore a lot more to always get the same few results.